机器人登录服务器优选bug

<!--
  感谢你来到这里,
  在反馈前, 请确认你已经做了下面这些事情
  - 对照 releases，相关问题未在近期更新中解决
  - 搜索了已有的 issues 列表中有没相关的信息
  - 阅读了 Mirai 的相关文档
-->

<!--在下面空白处详细描述你遇到的问题-->
机器人登录有时不会优先选择缓存的延迟低的服务器
而是从默认的服务器列表里随机选择
而且根据日志看貌似是尝试连接默认服务器之后再载入缓存的服务器列表
serever.json: https://pastebin.com/XrurseL4
可以看到并不包含183.232.94.44:80这个ip
这个ip是默认服务器列表里面的


<!--请尽量在 ``` 之间空白处附加全面的相关日志. 请不要截图. 如果日志过大, 可以上传文件.-->
```
[16:52:55] [DefaultDispatcher-worker-14/INFO]: [91m2021-10-23 16:52:55 W/Net 3528054353: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection timed out: no further information: /183.232.94.44:80
[16:52:55] [DefaultDispatcher-worker-14/INFO]: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection timed out: no further information: /183.232.94.44:80
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	Suppressed: kotlinx.coroutines.JobCancellationException: Parent job is Cancelling; job=JobImpl{Cancelling}@1f034ebf
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	[CIRCULAR REFERENCE:io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection timed out: no further information: /183.232.94.44:80]
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	[CIRCULAR REFERENCE:kotlinx.coroutines.JobCancellationException: Parent job is Cancelling; job=JobImpl{Cancelling}@1f034ebf]
[16:52:55] [DefaultDispatcher-worker-14/INFO]: Caused by: java.net.ConnectException: Connection timed out: no further information
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: 	at java.base/java.lang.Thread.run(Thread.java:834)
[16:52:55] [DefaultDispatcher-worker-14/INFO]: [0m
[16:53:19] [DefaultDispatcher-worker-17/INFO]: [0m2021-10-23 16:53:19 V/Net 3528054353: Loading server list from cache.[0m
[16:53:19] [DefaultDispatcher-worker-17/INFO]: [92m2021-10-23 16:53:19 I/Net 3528054353: Server list: 113.96.13.208:14000, 42.81.169.100:443, 42.81.184.206:443, 114.221.148.14:80, msfwifi.3g.qq.com:8080, 180.102.111.105:8080, 113.96.12.224:8080, 114.221.148.13:8080, 14.22.5.202:8080, 42.81.172.41:80.[0m
[16:53:19] [DefaultDispatcher-worker-17/INFO]: [92m2021-10-23 16:53:19 I/Bot 3528054353: Loaded account secrets from local cache.[0m
```


#### 复现
<!--在这里简略说明如何让这个问题再次发生-->
<!--可使用 1.  2.  3.  的列表格式，或其他任意恰当的格式-->
<!--如果你不确定如何复现, 请尽量描述发生当时的情景-->
重启整个机器人java程序登录,大概十几次会出现一次
我本地电脑测试有时候几乎是百分之50%的几率出现此情况
已经测试过2.8.0-M1 2.7.1 均有此情况
连接超时换服务器倒是还好，但是有时候会连接到延迟高的香港服务器
导致机器人登录缓慢，有时甚至几十分钟都载入不完群列表，导致接收不到聊天信息


#### 版本及相关选项选择

- mirai-core: `2.8.0-M1` <!--在 `` 中填写你正在使用的版本号，如 `2.6.7`-->
- bot-protocol: `ANDROID_PAD` <!-- 在 `` 中填写你正在使用的登录协议，如 `ANDROID_PHONE` -->

<!--如果你正在通过 mirai-console 或其他间接使用 mirai-core 的中间件（如 mirai-native、mirai-api-http 或者其他第三方 SDK）, 请同样提供版本号-->

#### 网络日志

<!--
请尽量在下方 ``` 空白处附加网络日志, 如果正在使用 mirai-console, 一般网络日志位于 bots/<****>/logs 里, 请不要截图.  如果日志过大, 可以上传文件
-->

```

```


#### 补充信息
<!--如有必要，你可以在下文继续添加其他信息-->

--- a/mirai-core/src/commonMain/kotlin/network/QQAndroidClient.kt
+++ b/mirai-core/src/commonMain/kotlin/network/QQAndroidClient.kt
@@ -39,7 +39,7 @@ internal fun getRandomByteArray(length: Int): ByteArray = ByteArray(length) { Ra
 // [114.221.148.179:14000, 113.96.13.125:8080, 14.22.3.51:8080, 42.81.172.207:443, 114.221.144.89:80, 125.94.60.148:14000, 42.81.192.226:443, 114.221.148.233:8080, msfwifi.3g.qq.com:8080, 42.81.172.22:80]
 
 internal val DefaultServerList: MutableSet<Pair<String, Int>> =
-    "msfwifi.3g.qq.com:8080, 14.215.138.110:8080, 113.96.12.224:8080, 157.255.13.77:14000, 120.232.18.27:443, 183.3.235.162:14000, 163.177.89.195:443, 183.232.94.44:80, 203.205.255.224:8080, 203.205.255.221:8080"
+    "msfwifi.3g.qq.com:8080"
         .split(", ")
         .map {
             val host = it.substringBefore(':')

--- a/mirai-core/src/commonMain/kotlin/network/components/ServerList.kt
+++ b/mirai-core/src/commonMain/kotlin/network/components/ServerList.kt
@@ -86,10 +86,7 @@ internal interface ServerList {
 
     companion object : ComponentKey<ServerList> {
         val DEFAULT_SERVER_LIST: Set<ServerAddress> =
-            """msfwifi.3g.qq.com:8080, 14.215.138.110:8080, 113.96.12.224:8080,
-                |157.255.13.77:14000, 120.232.18.27:443, 
-                |183.3.235.162:14000, 163.177.89.195:443, 183.232.94.44:80, 
-                |203.205.255.224:8080, 203.205.255.221:8080""".trimMargin()
+            """msfwifi.3g.qq.com:8080""".trimMargin()
                 .splitToSequence(",").filterNot(String::isBlank)
                 .map { it.trim() }
                 .map {

